"Symbol not found" error with ruby + jemalloc + clang

If build ruby which enables `jemalloc`, it will cause error in RMagick
 
 ## Reproduce
 
 ### 1. Compile Ruby with jemalloc
 ```
 $ CONFIGURE_OPTS="--with-jemalloc --disable-install-rdoc" MAKE_OPTS="-j 4" rbenv install 2.5.7
 ```
 
 ### 2. Run test
 ```
 $ ruby -v
ruby 2.5.7p206 (2019-10-01 revision 67816) [x86_64-darwin19]

$ rake
mkdir -p tmp/x86_64-darwin19/RMagick2/2.5.7
cd tmp/x86_64-darwin19/RMagick2/2.5.7
/Users/watson/.rbenv/versions/2.5.7/bin/ruby -I. ../../../../ext/RMagick/extconf.rb
checking for brew... yes
checking for clang... yes
checking for pkg-config... yes
checking for outdated ImageMagick version (<= 6.7.7)... no
Usage: file [bcCdEhikLlNnprsvzZ0] [-e test] [-f namefile] [-F separator] [-m magicfiles] [-M magicfiles] file...
       file -C -m magicfiles
Try `file --help' for more information.
checking for __GNUC__... yes
checking for Ruby version >= 2.3.0... yes
checking for MagickCore/MagickCore.h... no
checking for magick/MagickCore.h... yes
checking for GetImageChannelEntropy() in assert.h,ctype.h,stdio.h,stdlib.h,math.h,time.h,sys/types.h,magick/MagickCore.h... yes
checking for SetImageGray() in assert.h,ctype.h,stdio.h,stdlib.h,math.h,time.h,sys/types.h,magick/MagickCore.h... yes
creating extconf.h
creating Makefile


======================================================================
Sun 13 Oct 19 15:36:26
This installation of RMagick 4.1.0.rc2 is configured for
Ruby 2.5.7 (x86_64-darwin19) and ImageMagick 6.9.10
======================================================================


Configured compile options: {:magick_version=>"6.9.10", :local_libs=>" -L/Users/watson/imagemagick6.9/lib -lMagickCore-6.Q16", :cflags=>" -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/Users/watson/imagemagick6.9/include/ImageMagick-6 -std=gnu99", :cppflags=>"-I/usr/local/opt/libxml2/include -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/Users/watson/imagemagick6.9/include/ImageMagick-6", :ldflags=>"-L/usr/local/opt/libxml2/lib -L/Users/watson/imagemagick6.9/lib -lMagickCore-6.Q16", :defs=>[], :config_h=>"Makefile rmagick.h"}
cd -
cd tmp/x86_64-darwin19/RMagick2/2.5.7
/usr/bin/make
compiling ../../../../ext/RMagick/rmagick.c
compiling ../../../../ext/RMagick/rmdraw.c
compiling ../../../../ext/RMagick/rmenum.c
compiling ../../../../ext/RMagick/rmfill.c
compiling ../../../../ext/RMagick/rmilist.c
compiling ../../../../ext/RMagick/rmimage.c
../../../../ext/RMagick/rmimage.c:13822:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in
      C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Image, start_loop, bool)
^
../../../../ext/RMagick/rmagick.h:481:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
../../../../ext/RMagick/rmagick.h:438:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:89:1: note: expanded from here
C__Bool_to_R__Bool
^
../../../../ext/RMagick/rmimage.c:13822:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in
      C99 [-Wimplicit-function-declaration]
../../../../ext/RMagick/rmagick.h:482:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
../../../../ext/RMagick/rmagick.h:461:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:95:1: note: expanded from here
R__Bool_to_C__Bool
^
../../../../ext/RMagick/rmimage.c:16365:21: warning: enumeration value 'StaticGravity' not handled in switch [-Wswitch]
            switch (gravity)
                    ^
../../../../ext/RMagick/rmimage.c:16365:21: note: add missing switch cases
            switch (gravity)
                    ^
3 warnings generated.
compiling ../../../../ext/RMagick/rminfo.c
../../../../ext/RMagick/rminfo.c:244:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in C99
      [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Info, antialias, bool)
^
../../../../ext/RMagick/rmagick.h:481:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
../../../../ext/RMagick/rmagick.h:438:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:134:1: note: expanded from here
C__Bool_to_R__Bool
^
../../../../ext/RMagick/rminfo.c:244:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in C99
      [-Wimplicit-function-declaration]
../../../../ext/RMagick/rmagick.h:482:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
../../../../ext/RMagick/rmagick.h:461:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:140:1: note: expanded from here
R__Bool_to_C__Bool
^
../../../../ext/RMagick/rminfo.c:1135:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in
      C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Info, dither, bool)
^
../../../../ext/RMagick/rmagick.h:481:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
../../../../ext/RMagick/rmagick.h:438:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:166:1: note: expanded from here
C__Bool_to_R__Bool
^
../../../../ext/RMagick/rminfo.c:1135:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in
      C99 [-Wimplicit-function-declaration]
../../../../ext/RMagick/rmagick.h:482:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
../../../../ext/RMagick/rmagick.h:461:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:172:1: note: expanded from here
R__Bool_to_C__Bool
^
../../../../ext/RMagick/rminfo.c:1760:1: warning: implicit declaration of function 'C__Bool_to_R__Bool' is invalid in
      C99 [-Wimplicit-function-declaration]
DEF_ATTR_ACCESSOR(Info, monochrome, bool)
^
../../../../ext/RMagick/rmagick.h:481:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_READER(class, attr, type)\
    ^
../../../../ext/RMagick/rmagick.h:438:16: note: expanded from macro 'DEF_ATTR_READER'
        return C_##type##_to_R_##type(ptr->attr);\
               ^
<scratch space>:205:1: note: expanded from here
C__Bool_to_R__Bool
^
../../../../ext/RMagick/rminfo.c:1760:1: warning: implicit declaration of function 'R__Bool_to_C__Bool' is invalid in
      C99 [-Wimplicit-function-declaration]
../../../../ext/RMagick/rmagick.h:482:5: note: expanded from macro 'DEF_ATTR_ACCESSOR'
    DEF_ATTR_WRITER(class, attr, type)
    ^
../../../../ext/RMagick/rmagick.h:461:21: note: expanded from macro 'DEF_ATTR_WRITER'
        ptr->attr = R_##type##_to_C_##type(val);\
                    ^
<scratch space>:211:1: note: expanded from here
R__Bool_to_C__Bool
^
6 warnings generated.
compiling ../../../../ext/RMagick/rmkinfo.c
compiling ../../../../ext/RMagick/rmmain.c
compiling ../../../../ext/RMagick/rmmontage.c
compiling ../../../../ext/RMagick/rmpixel.c
compiling ../../../../ext/RMagick/rmstruct.c
compiling ../../../../ext/RMagick/rmutil.c
linking shared-object RMagick2.bundle
cd -
install -c tmp/x86_64-darwin19/RMagick2/2.5.7/RMagick2.bundle lib/RMagick2.bundle
cp tmp/x86_64-darwin19/RMagick2/2.5.7/RMagick2.bundle tmp/x86_64-darwin19/stage/lib/RMagick2.bundle
/Users/watson/.rbenv/versions/2.5.7/bin/ruby -I/Users/watson/.rbenv/versions/2.5.7/lib/ruby/gems/2.5.0/gems/rspec-support-3.9.0/lib:/Users/watson/.rbenv/versions/2.5.7/lib/ruby/gems/2.5.0/gems/rspec-core-3.9.0/lib /Users/watson/.rbenv/versions/2.5.7/lib/ruby/gems/2.5.0/gems/rspec-core-3.9.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb

Randomized with seed 12990
........................................*.......**...

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Magick::Draw#marshal_dump #marshal_load marshals without an error
     # this spec fails on some versions of ImageMagick
     # ./spec/rmagick/draw_spec.rb:82

  2) Magick::Image#read issue #200 not hangs with nil argument
     # No reason given
     # ./spec/rmagick/image/read_spec.rb:17

  3) Magick::Image#read issue #200 raise error with nil argument
     # No reason given
     # ./spec/rmagick/image/read_spec.rb:22


Finished in 2.21 seconds (files took 0.33171 seconds to load)
53 examples, 0 failures, 3 pending

Randomized with seed 12990

Loaded suite /Users/watson/.rbenv/versions/2.5.7/lib/ruby/gems/2.5.0/gems/rake-13.0.0/lib/rake/rake_test_loader
Started
........................................................................................................................
/Users/watson/prj/rmagick/test/Image2.rb:132: warning: CompressImageColormap failed (probably DirectClass image)
........................................................................................................................
.................................................................................................................../Users/watson/prj/rmagick/test/ImageList2.rb:257: warning: RMagick: animation only contains zero time delays `/Users/watson/prj/rmagick/doc/ex/images/Button_0.gif' @ warning/layer.c/RemoveZeroDelayLayers/1742
.....
..................................................dyld: lazy symbol binding failed: Symbol not found: _C__Bool_to_R__Bool
  Referenced from: /Users/watson/prj/rmagick/lib/RMagick2.bundle
  Expected in: flat namespace

dyld: Symbol not found: _C__Bool_to_R__Bool
  Referenced from: /Users/watson/prj/rmagick/lib/RMagick2.bundle
  Expected in: flat namespace

rake aborted!
SignalException: SIGABRT

Tasks: TOP => default => test
(See full trace by running task with --trace)
```

## Explanation

If enables `jemalloc`, it will include `stdbool.h`
https://github.com/jemalloc/jemalloc/blob/4094b7c03fb5e814f6f4c85ff7e93b3228dc4d29/include/jemalloc/jemalloc_macros.h.in#L2

Then,  `stdbool.h` will replace `bool` to `_Bool`.

```
$ cat /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/tapi/11.0.0/include/stdbool.h | grep "_Bool"
#define bool _Bool
/* Define _Bool as a GNU extension. */
#define _Bool bool
```

Finally, it will causes symbol error.

--- a/ext/RMagick/rmagick.h
+++ b/ext/RMagick/rmagick.h
@@ -414,8 +414,8 @@ EXTERN ID rm_ID_y;                 /**< "y" */
 /*
     Define simple attribute accessor methods (boolean, int, string, and double types)
 */
-#define C_bool_to_R_bool(attr) (attr) ? Qtrue : Qfalse /**< C boolean -> Ruby boolean */
-#define R_bool_to_C_bool(attr) RTEST(attr) /**<  C boolean <- Ruby boolean */
+#define C_boolean_to_R_boolean(attr) (attr) ? Qtrue : Qfalse /**< C boolean -> Ruby boolean */
+#define R_boolean_to_C_boolean(attr) RTEST(attr) /**<  C boolean <- Ruby boolean */
 #define C_int_to_R_int(attr) INT2FIX(attr) /**< C int -> Ruby int */
 #define R_int_to_C_int(attr) NUM2INT(attr) /**< C int <- Ruby int */
 #define C_long_to_R_long(attr) INT2NUM(attr) /**< C long -> Ruby long */

--- a/ext/RMagick/rmimage.c
+++ b/ext/RMagick/rmimage.c
@@ -13819,7 +13819,7 @@ Image_spread(int argc, VALUE *argv, VALUE self)
 }
 
 
-DEF_ATTR_ACCESSOR(Image, start_loop, bool)
+DEF_ATTR_ACCESSOR(Image, start_loop, boolean)
 
 
 /**

--- a/ext/RMagick/rminfo.c
+++ b/ext/RMagick/rminfo.c
@@ -241,7 +241,7 @@ static char *pixel_packet_to_hexname(PixelPacket *pp, char *name)
 #endif
 
 
-DEF_ATTR_ACCESSOR(Info, antialias, bool)
+DEF_ATTR_ACCESSOR(Info, antialias, boolean)
 
 /** Maximum length of a format (@see Info_aref) */
 #define MAX_FORMAT_LEN 60
@@ -1132,7 +1132,7 @@ Info_dispose_eq(VALUE self, VALUE disp)
     return disp;
 }
 
-DEF_ATTR_ACCESSOR(Info, dither, bool)
+DEF_ATTR_ACCESSOR(Info, dither, boolean)
 
 
 /**
@@ -1757,7 +1757,7 @@ Info_monitor_eq(VALUE self, VALUE monitor)
 
 
 
-DEF_ATTR_ACCESSOR(Info, monochrome, bool)
+DEF_ATTR_ACCESSOR(Info, monochrome, boolean)
 
 DEF_ATTR_ACCESSOR(Info, number_scenes, ulong)
 

